google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionClient.java

```diff
@@ -49,10 +51,16 @@ private SessionId(DatabaseId db, String name) {
     static SessionId of(String name) {
       Preconditions.checkNotNull(name);
       Map<String, String> parts = NAME_TEMPLATE.match(name);
+      if (parts == null) {
+        parts = EXTERNAL_HOST_NAME_TEMPLATE.match(name);
```

Comment:
Can you add unit tests to cover this pattern and ensure the functionality works as intended?

Reply:
Added unit test to cover existing pattern (for regression) as well as new pattern